user_sync/config.py

```diff
@@ -830,28 +829,37 @@ def get_credential(self, name, user_name, none_allowed=False):
                 '%s: No value in secure storage for user "%s", key "%s"' % (scope, user_name, secure_value_key))
         return value
 
-    def get_value_from_keyring(self, secure_value_key, user_name):
-        logger = logging.getLogger("keyring")
-        backend_list = {
-            'KWallet': 'keyring.backends.kwallet',
-            'SecretService': 'keyring.backends.SecretService',
-            'Windows': 'keyring.backends.Windows',
-            'chainer': 'keyring.backends.chainer',
-            'macOS': 'keyring.backends.OS_X',
-        }
-
-        for k, v in six.iteritems(backend_list):
-            logger.debug('Loading backend: ' + k)
-            suppress_exceptions(import_module(v))
-
-        viable_classes = KeyringBackend.get_viable_backends()
-        rings = list(suppress_exceptions(viable_classes, exceptions=TypeError))
-        selected = max(rings, default=fail.Keyring(), key=lambda p: p.priority)
-        keyring.set_keyring(selected)
-
-        logger.info("Using keyring '" + selected.name +  "' to retrieve: " + secure_value_key)
+    @staticmethod
+    def get_value_from_keyring(secure_value_key, user_name):
+        import keyring
+        if flags.get_flag('UST_BUILD_EXE'):
+            from importlib import import_module
+            from keyring.backends import fail
+            from keyring.backend import KeyringBackend
+            from keyring.util import suppress_exceptions
+
+            logger = logging.getLogger("keyring")
+            backend_list = {
+                'KWallet': 'keyring.backends.kwallet',
+                'SecretService': 'keyring.backends.SecretService',
+                'Windows': 'keyring.backends.Windows',
+                'chainer': 'keyring.backends.chainer',
+                'macOS': 'keyring.backends.OS_X',
+            }
+
+            for k, v in six.iteritems(backend_list):
+                logger.debug('Loading backend: ' + k)
+                suppress_exceptions(import_module(v))
+
+            viable_classes = KeyringBackend.get_viable_backends()
+            rings = list(suppress_exceptions(viable_classes, exceptions=TypeError))
+            selected = max(rings, default=fail.Keyring(), key=lambda p: p.priority)
+            keyring.set_keyring(selected)
+
+            logger.info("Using keyring '" + selected.name + "' to retrieve: " + secure_value_key)
```

Comment:
If we wanted to retain the message informing the user which attributes were being fetched, we could move the getlogger statement outside the conditional, and change line 859 to read:
logger.info("Using keyring '" + keyring.get_keyring().name +  "' to retrieve: " + secure_value_key)

But otherwise, this all looks great! Approved!